### ISSUE TITLE
HexSeq & HexSeqP ergonomics (keys & visibility)

### ISSUE THREAD

**Issue by firolightfog** [docb/dbRackModules#7]
Dear Christian,

I had to realize how powerful and cool these modules are. Using them I also identified a few minor things that could speed up my work with them. Let me share theses ideas with you for your consideration. 

- Change: color of 'modified but unsaved' sequence string should be more visible (#FF0000 or similar)
- Add: highlight background for focused sequence window (#FFFCCC or similar)

- Change: focusing a sequence window automatically selects the sequence string
- Add: allowing 'abcdef' keys (beside the currently recognized 'FEDCBA9876543210')
- Add: allowing up/down arrows to save the sequence string and move the focus up/down 
- Add: if no sequence window is selected then pressing key 1-9 focuses sequence windows 1-9 
- Add: pressing Tab focuses the next sequence window (I'm not sure if Tab is available for the modules tbh)

Best regards, Andras

**Comment by docb** [comment]
hi Andras, thanks for the suggestions:
- Change: color of 'modified but unsaved' sequence string should be more visible (#FF0000 or similar)
   Ok.

- Add: highlight background for focused sequence window (#FFFCCC or similar)
    Ok

- Change: focusing a sequence window automatically selects the sequence string
     Not sure. As i can currently click on the exact position i would want to change some thing and may not want to have another   click to deselect. (i dislike this feature in many other apps too).

- Add: allowing 'abcdef' keys (beside the currently recognized 'FEDCBA9876543210')
    Not sure. Currently i planned the lowercase chars for some functionality. 
    e.g. 'r' randomizes a field, 's' shifts a field by one bit, and i don't want to interfere with hex chars. 

- Add: allowing up/down arrows to save the sequence string and move the focus up/down
    Ok.    
    
- Add: if no sequence window is selected then pressing key 1-9 focuses sequence windows 1-9
   Don't know how this is possible as then i would be required to have a focused state on the module.
   Even if i click on the module and press 2 then the MIDI CV computer keyboard fires if configured

- Add: pressing Tab focuses the next sequence window (I'm not sure if Tab is available for the modules tbh)
    I can check this. But it would be the same as arrow down?

**Comment by firolightfog** [comment]
Hello Christian,

Thanks for considering the points! I'm glad that a few of them made it to your future development list. :) 

-----
I read your comments and I feel strongly about one point only: 

```
Add: allowing abcdef keys (beside the currently recognized FEDCBA9876543210)
Not sure. Currently i planned the lowercase chars for some functionality.
e.g. r randomizes a field, s shifts a field by one bit, and i dont want to interfere with hex chars.
```
I excitedly welcome any coming functions! I still think that at least 'abcdef' should be dedicated to the sequence windows and the hex strings. Switching the CapsLock before editing is weird and unusual. Hopefully you'll reconsider your opinion.

----

That's it. I'm happy even if you drop the rest of the suggestions. They would be improvements (imho) but wouldn't add to the user experience too much. 

```
Change: focusing a sequence window automatically selects the sequence string
Not sure. As i can currently click on the exact position ...
```
I rarely manage to click to the exact position I'd like to change. As I use the arrow keys to get to the right position, it seemed to be more intuitive for me. But OK, I see your point. 

```
Add: if no sequence window is selected then pressing key 1-9 focuses sequence windows 1-9
Don't know how this is possible as then i would be ...
```
I've seen something similar in [Voxglitch](https://github.com/clone45/voxglitch) [Digital Sequencer ](https://library.vcvrack.com/voxglitch/digitalsequencer) and found it handy. But OK...
(Actually if it's working then HexSeqP pattern select feature could benefit from it more.)

```
Add: pressing Tab focuses the next sequence window (I'm not sure if Tab is available for the 
modules tbh)
I can check this. But it would be the same as arrow down?
```
Arrow is suggested to 'save & jump'. Tab would 'jump only'. But probably it's not worth the effort. I just find myself unintentionally press Tab to change, so I though to get the most from it. ;) 

Best regards, A.

**Comment by firolightfog** [comment]
Suggestions were reviewed and related changes were implemented in v2.0.4
